source/common/common/hash.cc

```diff
@@ -0,0 +1,55 @@
+#include "common/common/hash.h"
+
+#include "absl/strings/string_view.h"
+
+namespace {
+
+static inline std::uint64_t unaligned_load(const char* p) {
```

Comment:
Please put a comment in this file also as to where all of this code was copied from, and why it has been copied.